src/gpio.rs

```diff
@@ -121,8 +121,8 @@ where
         self.pin
     }
 
-    fn reset(&mut self) -> Result<(), Self::Error> {
-        Ok(())
+    fn reset(&self) -> Result<(), Self::Error> {
+        esp_result!(unsafe { gpio_reset_pin(self.pin) }, ())
```

Comment:
We need to guard this with `#[cfg(not(feature = "ulp"))]` As I don't think there is a `gpio_reset_pin` in the "mini" HAL.

Reply:
It was my mistake that the CI does not have a build with `--features ulp`, but I just added it.

Reply:
And then, yeah, I feel `&mut` is kinda OK here :)